src/GraphQL/Definitions.hs

```diff
@@ -60,17 +60,13 @@ instance forall a as. (HasFieldDefinition a, HasFieldDefinitions as) => HasField
 instance HasFieldDefinitions '[] where
   getFieldDefinitions = []
 
--- Convert a type-level list of KnownSymbol into a value-level
--- list. This is used for Enums.
-class GetSymbolList a where
-  getSymbolList :: [Text]
-
-instance forall a as. (KnownSymbol a, GetSymbolList as) => GetSymbolList (a:as) where
-  getSymbolList = (toS (symbolVal (Proxy :: Proxy a))):(getSymbolList @as)
-
-instance GetSymbolList '[] where
-  getSymbolList = []
-
+-- | For each enum type we need 1) a list of all possible values 2) a
+-- way to serialise and 3) deserialise.
+class GraphQLEnum a where
```

Comment:
Yep, `Enum` type class is not what we want. That's used to do e.g. [Dog..Cat] (like [1..10]), max and min bounds. What we need from the enum are the three things I mentioned in the comment.